### PR TITLE
pillar: containerd: Create a pipe for container's stdin

### PR DESCRIPTION
**Pull request by rene**
Currently the device /dev/null is used as the standard input for a container task. However, any process reading from /dev/null will always get 0 bytes as the result, which makes interactive containers, e.g., those that runs a shell as the entrypoint (and expects data on the standard input), to exit right after starting. In order to keep the container task blocked (in these situations), a valid standard input device must be provided.  This commit creates a fake stdin that reads from an unpopulated pipe so it blocks the reader process.

This issue is not observed on standard containers because qemu's process doesn't expect any inputs on stdin. It only affects native containers.

### CODE REVIEW

File: pkg/pillar/containerd/containerd.go [hunk 1]
```diff
@@ -481,19 +481,23 @@ func (client *Client) CtrContainerInfo(ctx context.Context, name string) (int, i
...
+	if err != nil {
+		logrus.Errorf("CtrLogIOCreator: Error opening file %s with: %s", stdoutFile, err)
+	}
+	pipeStderr, err := os.OpenFile(stderrFile, os.O_WRONLY, 0)
+	if err != nil {
+		logrus.Errorf("CtrLogIOCreator: Error opening file %s with: %s", stderrFile, err)
 	}
 
+	// Create a fake stdin, so we won't break any interactive application
+	fakeStdin, _ := io.Pipe()
```
Comment (OhmSpectator) [inline pkg/pillar/containerd/containerd.go:497]: I'm just curious: how do we expect to work with interactive applications in containers, if we are not going to provide them any input...

**Review by OhmSpectator — APPROVED** [review]
Looks fine, although I'm not sure about the use case.

File: pkg/pillar/containerd/containerd.go (hunk 1 above)
Comment (rene) [reply pkg/pillar/containerd/containerd.go:497]: @OhmSpectator I experienced this issue while deploying a native standard CUDA container directly from docker hub (i.e., without any changes). The container can still be accessed, for example, using ctr command, the "fake" stdin will just avoid the container task to finalize and generate an error, as the Edge App had crashed....

**Comment by OhmSpectator** [comment]
It looks fine; I just want to see the Eden tests run.
@yash-zededa, could you please check why they are skipped here?

**Review by OhmSpectator — APPROVED** [review]

**Review by deitch — APPROVED** [review]
And it is easier to read as well.

**Pull request merged**